### PR TITLE
Asset integration - Adding tabs for Select Timeseries, Select Timeseries from Asset, Custom Query

### PR DESCRIPTION
**Pull request by hulien22**

### CODE REVIEW

File: src/datasource.ts [hunk 1]
```diff
@@ -264,25 +321,156 @@ export default class CogniteDatasource {
...
+    }
+    if (filterOptions.granularity) {
+      target.granularity = filterOptions.granularity;
+    }
+  }
+
+  parse(customQuery) {
+    // replace variables with their values
+    for (let templateVariable of this.templateSrv.variables) {
+      customQuery = customQuery.replace("[[" + templateVariable.name + "]]", templateVariable.current.value);
```
[the remark this answers is not in the document]
Comment (martincognite) [reply src/datasource.ts]: `customQuery.replace(/(\$\w+)/, 'my_asset')` would do the trick for substitution.

File: src/datasource.ts [hunk 2]
```diff
@@ -264,25 +321,156 @@ export default class CogniteDatasource {
...
+    }
+
+    let filtersOptions = {
+      filters: [],
+      granularity: '',
+      aggregation: ''
+    };
+
+    const timeseriesRegex = /^timeseries\{(.*)\}(?:\[(.*)\])$/;
```
Comment (martincognite) [inline src/datasource.ts]: I don't have a good alternative solution for the time being, but this section looks like a nightmare to maintain 👿

File: src/datasource.ts [hunk 3]
```diff
@@ -264,25 +321,156 @@ export default class CogniteDatasource {
...
+
     return this.backendSrv.datasourceRequest({
       url: this.url + urlEnd,
       method: "GET",
     }).then((result: { data: TimeSeriesResponse }) =>
       result.data.data.items.map(timeSeriesResponseItem => (
         {
           text: timeSeriesResponseItem.name,
-          value: timeSeriesResponseItem.name
+          value: (type==="Asset") ? '' + timeSeriesResponseItem.id : timeSeriesResponseItem.name
```
[the remark this answers is not in the document]
Comment (hulien22) [reply src/datasource.ts]: Good point, added in an enum for that

File: src/datasource.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (hulien22) [reply src/datasource.ts]: Added.

File: src/datasource.ts (hunk 2 above)
Comment (hulien22) [reply src/datasource.ts]: Added a comment to describe what this regex is doing. 
I think for the time being this is a fine way to approach this, but if we want to add more complexity to the querying, then I agree that we will have to change some things.

**Pull request merged**